Packet++/src/SomeIpLayer.cpp

```diff
@@ -269,14 +270,25 @@ bool SomeIpLayer::subProtocolFollows() const
 		return true;
 	}
 	/* Check if the payload is a valid SOME/IP-TP layer */
-	if (isSomeIpTpPacket(getMessageTypeAsInt()))
+	if (SomeIpTpLayer::isSomeIpTpPacket(getMessageTypeAsInt()))
 	{
 		return true;
 	}
 
 	return false;
 }
 
+void SomeIpLayer::computeCalculateFields()
+{
+	if (!m_NextLayer)
+		return;
+
+	if ((m_NextLayer->getProtocol() == SomeIPTP)) // TODO: handling of SomeIpSd
+	{
+		setPayloadLength(m_NextLayer->getHeaderLen());
+	}

```

Comment:
Before making any other comments, I wanted to discuss the design decision to make SOME/IP TP a layer on top of SOME/IP, so basically every TP packet will look like this:

```text
--------------------
| Eth layer        |
--------------------
| IPv4 layer       |
--------------------
| UDP layer        |
--------------------
| SOME/IP layer    |
--------------------
| SOME/IP TP layer |
--------------------
```

An alternative could be that `SomeIpTpLayer` inherits from `SomeIpLayer` and then the same packet will look like this: 

```text
--------------------
| Eth layer        |
--------------------
| IPv4 layer       |
--------------------
| UDP layer        |
--------------------
| SOME/IP TP layer |
--------------------
```

I'm not an expert in SOME/IP, so I'll need your guidance here: which of the designs makes more sense in terms of usage and user experience?

Reply:
it should be ok with both designs, maybe the handling is a little bit easier with one layer but i have not really a preference. What do you think which one we should use?

Reply:
I agree, I think if we have the TP layer inherit from Some/IP layer it will be easier to manage. So basically the second option of the 2 I proposed. Will you make the relevant changes?

Reply:
Yes, currently working on it.

Reply:
currently updated, can you please take a look on this PR?

Reply:
Sure, thanks @kolbex ! I'll review it soon